ts/src/decoding/fastPforUnpack.ts

```diff
@@ -0,0 +1,844 @@
+/**
+ * Optimized bit-unpacking routines for FastPFOR decoding.
+ * The functions are mechanically structured; treat edits as a batch.
+ */
+
+import { MASKS } from "./fastPforSpec";
+
+export function fastUnpack32_1(inValues: Int32Array, inPos: number, out: Int32Array, outPos: number): void {
+    const in0 = inValues[inPos] >>> 0;
+    for (let i = 0; i < 32; i++) {
+        out[outPos + i] = (in0 >>> i) & 1;
+    }
+}
+
+export function fastUnpack32_2(inValues: Int32Array, inPos: number, out: Int32Array, outPos: number): void {
```

Comment:
These methods look very similar, consider creating a script that generates these methods as part of the "build" process in this repo to avoid manual mistakes...